astroid/interpreter/_import/spec.py

```diff
@@ -359,6 +367,28 @@ def _find_spec_with_path(
             continue
         return finder_instance, spec
 
+    # Support for custom finders
+    for meta_finder in sys.meta_path:
+        spec = meta_finder.find_spec(modname, submodule_path)
```

Comment:
I think we have to guard against `find_spec` being absent. It was only recently added to `six._SixMetaPathImporter`, see https://github.com/benjaminp/six/pull/352.

Reply:
Hm, this is an issue within `six` then..

https://github.com/python/typeshed/blob/b9270da7b083f08a86e4366b3cf284e627a63873/stdlib/sys.pyi#L45

The `stdlib` is typed with a protocol to signal that this should indeed inherit from the `abc`. I'll make us use a `Protocol` as well.

Reply:
I think part of the issue is that PEP 302 doesn't mention `find_spec`, so there could be code in the wild without it.

Reply:
See my last commit, would that be an okay solution?

Reply:
I guess I still don't understand, an importer without a find_spec method, such as [older vendored versions of six](https://sourcegraph.com/github.com/cronyo/cronyo/-/blob/cronyo/vendor/urllib3/packages/six.py?subtree=true) will still raise an uncaught AttributeError on `spec = meta_finder.find_spec(modname, submodule_path)`.

Reply:
The issue is that `find_module` has been deprecated since 3.4:
https://docs.python.org/3/library/importlib.html#importlib.abc.MetaPathFinder.find_module

Anyway, I'll keep the typing and use a `hasattr`. That seems the most "safe" while also correct.